### PR TITLE
geo/geomfn: implement ST_MemSize({geometry})

### PR DESCRIPTION
**Pull request by C0rWin**
This commit add ST_MemSize() builtin which returns the amount of
memory space (in bytes) the geometry takes.

Signed-off-by: Artem Barger <bartem@il.ibm.com>

Release note (sql change): Adds ST_MemSize to return memory space the
geometry takes.

### CODE REVIEW

**Comment by C0rWin** [comment]
Addresses #48985.

File: pkg/sql/logictest/testdata/logic_test/geospatial [hunk 1]
```diff
@@ -5392,3 +5392,46 @@ SRID=4326;POLYGON EMPTY                  SRID=4326;MULTIPOLYGON EMPTY
...
 SRID=4004;LINESTRING (-180 -90, 180 90)  SRID=4004;MULTIPOLYGON (((45 35.264389682754654, 135 35.264389682754654, -135 35.264389682754654, -45 35.264389682754654, 45 35.264389682754654)), ((135 35.264389682754654, 135 -35.264389682754654, -135 -35.264389682754654, -135 35.264389682754654, 135 35.264389682754654)), ((-135 -35.264389682754654, 135 -35.264389682754654, 45 -35.264389682754654, -45 -35.264389682754654, -135 -35.264389682754654)))                                                                                                                                                                                                                    SRID=4004;MULTIPOLYGON (((45 35.264389682754654, 135 35.264389682754654, -135 35.264389682754654, -45 35.264389682754654, 45 35.264389682754654)), ((135 35.264389682754654, 135 -35.264389682754654, -135 -35.264389682754654, -135 35.264389682754654, 135 35.264389682754654)), ((-135 -35.264389682754654, 135 -35.264389682754654, 45 -35.264389682754654, -45 -35.264389682754654, -135 -35.264389682754654)))
+
+
+subtest st_memsize
```
Comment (otan) [inline pkg/sql/logictest/testdata/logic_test/geospatial:5397]: can you simplify these tests:
```
SELECT ST_MemSize(g) FROM ( VALUES 
   (ST_GeomFromText('POINT EMPTY')),
   (ST_...),
) mem_size_test(g)
```

Comment (C0rWin) [reply pkg/sql/logictest/testdata/logic_test/geospatial:5397]: of course

**Review by otan — APPROVED** [review]

**Pull request merged**